lib/watir-webdriver/wait.rb

```diff
@@ -114,18 +107,44 @@ def respond_to?(*args)
       @element.respond_to?(*args)
     end
 
-    def method_missing(m, *args, &block)
+    def method_missing(decorator, m, *args, &block)
```

Comment:
I'm not sure it's a very good idea to have custom API for `method_missing` even though it's not designed to be used directly.

Maybe it's better to duck type the method that should be waited for? For example:

```ruby
class BaseDecorator
  def method_missing(m, *args, &block)
    Watir::Wait.until(@timeout, @message) { wait_until }
  end
end

class WhenPresentDecorator < BaseDecorator
  private
  def wait_until
    @element.present?
  end
end

class WhenEnabledDecorator < BaseDecorator
  private
  def wait_until
    @element.enabled?
  end
end
```

@titusfortner Any idea?